### PR TITLE
Dark/Night Theme

### PR DESCRIPTION
**Pull request by SachinRammoorthy**
Part of [this](https://codein.withgoogle.com/dashboard/task-instances/4627460274520064/) task.

- Also added sharedPref so the choice is saved even when the app is closed (see video).
- Visibility of 'Toggle theme' and the corresponding icon is also set depending on the choice.
- Theme can be changed either from **Nav Drawer** or the **Settings** page.
- Did not make use of `AppCompatDelegate`. Everything coded from scratch.

### [Video demo](https://youtu.be/hlx2BR8A1ww)

**Update**
- Can now toggle dark theme from Navigation View or Settings. **Please watch [Video demo](https://youtu.be/hlx2BR8A1ww).**
- Changed color scheme for dark theme.
- All text in dark mode is now white. Screenshot of color changes:

screen shot 2018-01-13 at 11 02 24 pm

PS: Minor issue in squashing commits. Cannot as commit history is mixed up after merging with upstream. Hope thats alright.

### CODE REVIEW

File: app/src/main/res/layout/settings_content_main.xml [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
+            android:layout_height="wrap_content"
+            android:id="@+id/switch1"
+            android:padding="17.5dp"
+            android:layout_marginLeft="175dp"/>
+
+        </LinearLayout>
+
+    </fragment>
+
 </LinearLayout>
```
Comment (sidhyatikku) [inline app/src/main/res/layout/settings_content_main.xml]: Blank line !

**Comment by SachinRammoorthy** [comment]
@aashutoshrathi have updated it. Please have a look. Here are screenshots of the changes.

screen shot 2018-01-13 at 8 50 58 pm
screen shot 2018-01-13 at 8 51 11 pm

**Comment by geek-gupta** [comment]
@SachinRammoorthy Add Contact activity theme doesn't change on toggle theme selection.

**Comment by geek-gupta** [comment]
And toggle button in setting activity is not applied to the app as whole. It only toggles the current(Settings) activity.

**Comment by SachinRammoorthy** [comment]
@geek-gupta have updated it. 

- The theme of Add Contact Activity also is set according to choice made either in Nav Drawer or Settings. 
- Toggle button in Settings already works for whole app...shown in video demo. Was it not working for you?

**Comment by geek-gupta** [comment]
@SachinRammoorthy let me check

**Comment by geek-gupta** [comment]
@SachinRammoorthy Working fine now

**Comment by geek-gupta** [comment]
@SachinRammoorthy good work :+1:

**Pull request merged**